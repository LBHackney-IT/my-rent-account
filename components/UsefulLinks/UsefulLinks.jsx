import React from "react";

import { links } from "data/usefulLinks.json";

const UsefulLinks = () => (
  <div className="govuk-body">
    <h2>Useful links</h2>
    {links.map(({ link, title }) => (
      <p key={link}>
        <a className="govuk-link" href={link}>
          {title}
        </a>
      </p>
    ))}
  </div>
);

export default UsefulLinks;
