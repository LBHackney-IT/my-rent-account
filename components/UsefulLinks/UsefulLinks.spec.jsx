import React from "react";
import { render } from "@testing-library/react";

import UsefulLinks from "./UsefulLinks";

jest.mock("data/usefulLinks.json", () => ({
  links: [
    {
      title: "foo",
      link: "https://hackney.gov.uk/foo/",
    },
    {
      title: "bar",
      link: "https://hackney.gov.uk/bar/",
    },
  ],
}));

describe("UsefulLinks component", () => {
  it("should render properly", () => {
    const { asFragment } = render(<UsefulLinks />);
    expect(asFragment()).toMatchSnapshot();
  });
});
