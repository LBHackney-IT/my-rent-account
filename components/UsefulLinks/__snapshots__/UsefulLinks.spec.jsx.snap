// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsefulLinks component should render properly 1`] = `
<DocumentFragment>
  <div
    class="govuk-body"
  >
    <h2>
      Useful links
    </h2>
    <p>
      <a
        class="govuk-link"
        href="https://hackney.gov.uk/foo/"
      >
        foo
      </a>
    </p>
    <p>
      <a
        class="govuk-link"
        href="https://hackney.gov.uk/bar/"
      >
        bar
      </a>
    </p>
  </div>
</DocumentFragment>
`;
